lib/vanagon/platform/windows.rb

```diff
@@ -53,9 +53,21 @@ def generate_packaging_artifacts(workdir, name, binding)
       # @param name [String] name of the project
       # @param binding [Binding] binding to use in evaluating the packaging templates
       def generate_msi_packaging_artifacts(workdir, name, binding)
-        FileUtils.mkdir_p(File.join(workdir, "wix"))
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.wxs.erb"), File.join(workdir, "wix",  "#{name}.wxs"), false, { :binding => binding })
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.filter.xslt.erb"), File.join(workdir, "wix", "#{name}.filter.xslt"), false, { :binding => binding })
+        # Copy (cp -r) files to workdir/wix folder. The VANAGON_ROOT root files are copied first, followed by the project specific files.
+        # The "remove_destination" flag is set which ensures that project specific files will always over-write generic/vanagon files
+        FileUtils.cp_r(File.join(VANAGON_ROOT, "resources/windows/wix"), workdir, :verbose => true, :remove_destination => true)
+        FileUtils.cp_r("resources/windows/wix", workdir, :verbose => true, :remove_destination => true)
+        # Evaluate and process all *.erb files found in workdir/wix
+        # The originating .erb file is left in place as it is ignored by the wix/candle process.
+        Dir.chdir(File.join(workdir, "wix")) do
+          Dir.glob("**/*.erb").each do |erb_filename|
+            target_file = File.basename(erb_filename, ".erb")
+            if File.exists?(target_file)
+              fail "Template file #{erb_filename} conflicts with static file #{target_file}"
```

Comment:
The fail cases make sense.

However, I'm trying to understand why you decided to break down the allow cases as you did. We have a set of generic files in vanagon. Some are static some are templates. Those will be used to build the MSI **unless** the equivalent of those files are provided by the project.

So, if the project provides properties.wxs OR properties.wxs.erb, it should override whatever equivalent file is provided by vanagon, be that properties.wxs or properties.wxs.erb.

We're not using both versions of the file if we have one from vanagon and one from the project.

Also, we shouldn't need to check for duplicate (both .wxs and .wxs.erb) files from vanagon because I trust the people developing vanagon to know better.